### ISSUE TITLE
Revert (parts of) #1005: default project to include everything (similar to `{"*":1}`)

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/data-api#1031]
Looks like downstream dependencies need time to adapt, before we can do #1005 change -- so let's revert part that actually changes the default projection; leaving parts that allow include/exclude of `$vector` (and `$vectorize`).
I will also create another issue for re-merging parts of #1005 once we have confidence it can be applied.
